File: caper/samples/ruby/hello1.rb
require './hello1_parser.rb'

class SemanticAction
    def syntax_error
    end

    def stack_overflow
    end
    
    def downcast v
        v
    end

    def upcast v
        v
    end
    
    def Greet
        print "hello world\n"
    end
end

parser = Hello_world::Parser.new(SemanticAction.new)

parser.post :token_Hello, 0
parser.post :token_World, 0
parser.post :token_eof, 0

File: caper/samples/php/hello1.php
<?php

error_reporting(E_ALL);

require_once('hello1_parser.php');

class SemanticAction
{
    function syntax_error()
    {
    }

    function stack_overflow()
    {
    }
    
    function downcast($y)
    {
        return $y;
    }

    function upcast($y)
    {
        return $y;
    }
    
    function Greet()
    {
        header('Content-Type: text/html');
        echo("hello world\n");
    }
}

$parser = new hello_world\Parser(new SemanticAction);

$parser->post(hello_world\Token::token_Hello, 0);
$parser->post(hello_world\Token::token_World, 0);
$parser->post(hello_world\Token::token_eof, 0);
